docs/source/cli/model-training-inference/single-machine-training-inference.rst

```diff
@@ -1,6 +1,74 @@
 
-Model Training and Inference on a Single Machine
+Model Training and Inference in a Single Machine
 -------------------------------------------------
+While the :ref:`Standalone Mode Quick Start <quick-start-standalone>` tutorial introduces some basic concepts, commands, and steps of using GprahStorm CLIs in a single machine, this user guide provides more detailed description of the usage of GraphStorm CLIs in a single machine. In addition, the majority of the descriptions in this guide can be directly applied to :ref:`model training and inference on distributed clusters <distributed-cluster>`.
 
-(Under Construction)
\ No newline at end of file
+GraphStorm can support graph machine learning (GML) model training and inference for common GML tasks, including **node classification**, **node regression**, **edge classification**, **edge regression**, and **link prediction**. For each task, GraphStorm provide a dedicated CLI for model training and inference. These CLIs share the same command template and some configurations, while each CLI has its unique task-specific configurations. GraphStorm also has a task-agnostic CLI for users to run your customized models.
+
+Task-specific CLI template for model training and inference
+............................................................
+GraphStorm model training and inference CLIs like the commands below. 
+
+.. code-block:: bash
+
+    # Model training
+    python -m graphstorm.run.TASK_COMMAND \
+              --num-trainers 1 \
+              --part-config data.json \
+              --cf config.yaml \
+              --save-model-path model_path/
+
+    # Model inference
+    python -m graphstorm.run.TASK_COMMAND \
+              --inference \
+              --num-trainers 1 \
+              --part-config data.json \
+              --cf config.yaml \
+              --restore-model-path model_path/ \
+              --save-prediction-path pred_path/
+
+In the above two CLIs, the ``TASK_COMMAND`` represents one of the five task-specific commands:
+
+    * ``gs_node_classification`` for node classification tasks;
+    * ``gs_node_regression`` for node regression tasks;
+    * ``gs_edge_classification`` for edge classification tasks;
+    * ``gs_edge_regression`` for edge regression tasks;
+    * ``gs_link_prediction`` for link prediction tasks.
+
+These task-specific commands work for both model training and inference except that inference CLI needs to add the **-\-inference** argument to indicate this is an inference CLI, and the **-\-restore-model-path** argument that indicates the path of the saved model checkpoint.
+
+For a single machine the argument **-\-num-trainers** can configure how many GPUs or CPU processes to be used. If using a GPU installed machine, the value of **-\-num-trainers** should be **equal or less** than the total number of available GPUs, while in a CPU-only machine, the value could be less than the total number of CPU processes to avoid errors.
```

Comment:
```suggestion
For a single machine, the argument **-\-num-trainers** can configure how many GPUs or CPU processes to be used. If using a GPU machine, the value of **-\-num-trainers** should be **equal or less** than the total number of available GPUs, while in a CPU-only machine, the value could be less than the total number of CPU processes to avoid errors.
```